e2e/pushpull_test.go

```diff
@@ -87,6 +95,103 @@ func runWithDindSwarmAndRegistry(t *testing.T, todo func(dindSwarmAndRegistryInf
 
 }
 
+func TestPushArchs(t *testing.T) {
+	runWithDindSwarmAndRegistry(t, func(info dindSwarmAndRegistryInfo) {
+		testCases := []struct {
+			name              string
+			args              []string
+			expectedPlatforms []manifestlist.PlatformSpec
+		}{
+			{
+				name: "default",
+				args: []string{},
+				expectedPlatforms: []manifestlist.PlatformSpec{
+					{
+						OS:           "linux",
+						Architecture: "amd64",
+					},
+				},
+			},
+			{
+				name: "all-platforms",
+				args: []string{"--all-platforms"},
+				expectedPlatforms: []manifestlist.PlatformSpec{
+					{
+						OS:           "linux",
+						Architecture: "amd64",
+					},
+					{
+						OS:           "linux",
+						Architecture: "386",
+					},
+					{
+						OS:           "linux",
+						Architecture: "ppc64le",
+					},
+					{
+						OS:           "linux",
+						Architecture: "s390x",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm",
+						Variant:      "v5",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm",
+						Variant:      "v6",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm",
+						Variant:      "v7",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm64",
+						Variant:      "v8",
+					},
+				},
+			},
+		}
+
+		for _, testCase := range testCases {
+			t.Run(testCase.name, func(t *testing.T) {
+				cmd := info.configuredCmd
+				ref := info.registryAddress + "/test/push-pull:1"
+				args := []string{"app", "push", "--tag", ref, "--insecure-registries=" + info.registryAddress}
+				args = append(args, testCase.args...)
+				args = append(args, filepath.Join("testdata", "push-pull", "push-pull.dockerapp"))
+				cmd.Command = dockerCli.Command(args...)
+				icmd.RunCmd(cmd).Assert(t, icmd.Success)
+
+				http := &http.Client{}
+				r, err := http.Get("http://" + info.registryAddress + "/v2/test/push-pull/manifests/1")
+				assert.NilError(t, err)
+				defer r.Body.Close()
+				assert.Equal(t, r.StatusCode, 200)
+				index := v1.Index{}
+				err = json.NewDecoder(r.Body).Decode(&index)
+				assert.NilError(t, err)
+				digest, err := getManifestListDigest(index)
+				assert.NilError(t, err)
+				r2, err := http.Get("http://" + info.registryAddress + "/v2/test/push-pull/manifests/" + digest.String())
+				assert.NilError(t, err)
+				defer r2.Body.Close()
+				assert.Equal(t, r2.StatusCode, 200)
+				manifestList := manifestlist.ManifestList{}
+				err = json.NewDecoder(r2.Body).Decode(&manifestList)
+				assert.NilError(t, err)
+				assert.Equal(t, len(manifestList.Manifests), len(testCase.expectedPlatforms), "Unexpected number of platforms")
+				for _, m := range manifestList.Manifests {
+					testify.Containsf(t, testCase.expectedPlatforms, m.Platform, "Platform expected but not found: %s", m.Platform)
```

Comment:
I don't think we need testify to do this. We can `gotest.tools` has the ability to do a `Contains` (search for `contains` on https://godoc.org/gotest.tools/assert).

Reply:
I haven't seen the function. Fixed